homeassistant/components/shelly/config_flow.py

```diff
@@ -182,3 +186,42 @@ async def _async_get_info(self, host):
                 aiohttp_client.async_get_clientsession(self.hass),
                 host,
             )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Define the config flow to handle options."""
+        return OptionsFlow(config_entry)
```

Comment:
We should only trigger the options flow if the device of the config entry supports it.

Reply:
Yeah that would be ideal, but I couldn't figure out how to set that up. Do you have any documentation to point to, or other integrations that work that way? I tried returning `None` but that just causes all sorts of errors.